ios/RNCSafeAreaProvider.m

```diff
@@ -1,19 +1,33 @@
 #import "RNCSafeAreaProvider.h"
 
+#import <React/RCTEventDispatcher.h>
 #import <React/RCTBridge.h>
 #import <React/RCTUIManager.h>
+#import "RCTUIManagerObserverCoordinator.h"
 #import "RNCSafeAreaUtils.h"
+#import "RNCChangeEvent.h"
+
+@interface RNCSafeAreaProvider () <RCTUIManagerObserver>
+
+@end
 
 @implementation RNCSafeAreaProvider {
+  id<RCTEventDispatcherProtocol> _eventDispatcher;
   UIEdgeInsets _currentSafeAreaInsets;
   CGRect _currentFrame;
   BOOL _initialInsetsSent;
+  uint16_t _coalescingKey;
```

Comment:
I can't see how this would ever be changed from the default of `0` - is it actually needed?